packages/discovery/src/discovery/config/ConfigReader.ts

```diff
@@ -119,4 +120,37 @@ export class ConfigReader {
 
     return projects
   }
+
+  async readRawConfigWithComments(
+    name: string,
+    chain: string,
+  ): Promise<RawDiscoveryConfig> {
+    assert(
+      fileExistsCaseSensitive(`discovery/${name}`),
+      'Project not found, check if case matches',
+    )
+    assert(
+      fileExistsCaseSensitive(`discovery/${name}/${chain}`),
+      'Chain not found in project, check if case matches',
+    )
+
+    const contents = await readFile(
+      `discovery/${name}/${chain}/config.jsonc`,
+      'utf-8',
+    )
+    const parsed: unknown = parseWithComments(contents)
+
+    // Parsing via Zod would effectively remove symbols and thus comments
+    assertDiscoveryConfig(parsed)
+
+    assert(parsed.chain === chain, 'Chain mismatch in config.jsonc')
+
+    return parsed
+  }
+}
```

Comment:
There is duplication of reading the config in here now. We should extract the common part, just call `readRawConfigWithComments` in `readConfig` and drop the comments there.